### PR TITLE
kb(treeview): raname node

### PR DESCRIPTION
**Pull request by marin-bratanov**

### CODE REVIEW

**Review by ntacheva — APPROVED** [review]

File: knowledge-base/treeview-rename-node.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: Is there a way to rename a node?
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:2]: More keywords:
```suggestion
title: How to Rename a TreeView Node
```

File: knowledge-base/treeview-rename-node.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+description: How to edit the name (text) of a treeview node
+type: how-to
+page_title: How to rename a node?
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:5]:
```suggestion
page_title: How to rename a TreeView node?
```

File: knowledge-base/treeview-rename-node.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+slug: treeview-kb-rename-node
+position: 
+tags:
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:8]:
```suggestion
tags: edit, rename, tree, treeview, note, item
```

File: knowledge-base/treeview-rename-node.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+</table>
+
+
+## Description
+Is there any way I can edit the node text within a Blazor TreeView? 
+
+I want to let the user rename a node.
+
+## Solution
+You can use the ItemTemplate to define how the node text will render and the desired UI and UX for editing. You can find an example in the following sample project: https://github.com/telerik/blazor-ui/tree/master/treeview/rename-node
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:30]: Add 1-2 sentences that say what the demo does.

```suggestion
You can use the `ItemTemplate` to define how the node text will render and the desired UI and UX for editing. Here is a sample project that demonstrates a possible implementation. It uses a custom component `EditableTreeNode` that renders buttons and a textbox to update the TreeView item.

https://github.com/telerik/blazor-ui/tree/master/treeview/rename-node
```

**Review by dimodi — APPROVED** [review]

**Pull request merged**